salt/master.py

```diff
@@ -1037,16 +1038,17 @@ def _post_stats(self, start, cmd):
         ) / self.stats[cmd]["runs"]
         if end - self.stat_clock > self.opts["master_stats_event_iter"]:
             # Fire the event with the stats and wipe the tracker
-            self.aes_funcs.event.fire_event(
-                {
-                    "time": end - self.stat_clock,
-                    "worker": self.name,
-                    "stats": self.stats,
-                },
-                tagify(self.name, "stats"),
+            stats_last = {
+                "time": end - self.stat_clock,
+                "worker": self.name,
+                "stats": self.stats,
+            }
+            self.aes_funcs.event.fire_event(stats_last, tagify(self.name, "stats"))
+            self.stats = collections.defaultdict(
+                lambda: {"mean": 0, "latency": 0, "runs": 0}
             )
-            self.stats = collections.defaultdict(lambda: {"mean": 0, "runs": 0})
             self.stat_clock = end
+            log.error("STATS %s", pprint.pformat(stats_last))
```

Comment:
I wonder if we want to error here. I know you weren't the original author so @dwoz do you still want to log error here. Seems a bit noisy